### PR TITLE
Migrating to Androidx

### PR DESCRIPTION
**Pull request by akhilamadari**
Migrating project to AndroidX

### CODE REVIEW

File: app/build.gradle [hunk 1]
```diff
@@ -127,47 +127,48 @@ androidExtensions {
 
 ext {
     // App dependency versions
-    supportVersion = '27.1.1'
-    playServicesVersion = '15.0.1'
-    daggerVersion = '2.14.1'
-    retrofitVersion = '2.3.0'
-    okHttpVersion = '3.9.1'
+    supportVersion = '1.0.0-beta01'
```
Comment (patrickhammond) [inline app/build.gradle]: We will probably want to drop this and move to individually versioning the support dependencies.

File: app/build.gradle [hunk 2]
```diff
@@ -127,47 +127,48 @@ androidExtensions {
...
-    implementation 'com.android.support.constraint:constraint-layout:1.1.2'
-    implementation "com.android.support:appcompat-v7:$supportVersion"
-    implementation "com.android.support:recyclerview-v7:$supportVersion"
-    implementation "com.android.support:cardview-v7:$supportVersion"
-    implementation "com.android.support:support-annotations:$supportVersion"
-    implementation "com.android.support:design:$supportVersion"
+    implementation 'androidx.core:core-ktx:1.0.1'
+    implementation 'androidx.constraintlayout:constraintlayout:1.1.3'
+    implementation "androidx.appcompat:appcompat:1.1.0-alpha02"
+    implementation "androidx.recyclerview:recyclerview:1.1.0-alpha02"
```
Comment (patrickhammond) [inline app/build.gradle]: If there is an option between the alpha dependency and a stable one, let's try to prefer the stable one for the dependencies.

File: app/build.gradle [hunk 3]
```diff
@@ -192,21 +193,26 @@ dependencies {
 
     // Monitoring - Leak Canary
     debugImplementation "com.squareup.leakcanary:leakcanary-android:$leakcanaryVersion"
+    debugImplementation 'androidx.fragment:fragment-testing:1.1.0-alpha04'
```
Comment (patrickhammond) [inline app/build.gradle]: Do we still need this dependency? If we do, we probably want to make sure it is scoped to `testImplementation`.

File: app/src/androidTest/java/com/mycompany/myapp/RecyclerViewMatcher.kt [hunk 4]
```diff
@@ -43,7 +43,7 @@ class RecyclerViewMatcher(private val recyclerViewId: Int) {
                 this.resources = view.resources
 
                 if (childView == null) {
-                    val recyclerView = view.rootView.findViewById<View>(recyclerViewId) as RecyclerView
+                    val recyclerView = view.rootView.findViewById<View>(recyclerViewId) as androidx.recyclerview.widget.RecyclerView
```
Comment (patrickhammond) [inline app/src/androidTest/java/com/mycompany/myapp/RecyclerViewMatcher.kt]: Can we fix the FQN (fully qualified names) of the classes throughout? The import should update and probably not the rest of the code.

File: app/src/debug/AndroidManifest.xml [hunk 5]
```diff
@@ -12,6 +12,7 @@
     <uses-permission android:name="android.permission.WAKE_LOCK"/>
 
     <!-- vmSafeMode skips the ART ahead of time compilation which can slow down development cycles -->
-    <application android:vmSafeMode="true"/>
+    <application android:vmSafeMode="true"
+        tools:ignore="GoogleAppIndexingWarning" />
```
Comment (patrickhammond) [inline app/src/debug/AndroidManifest.xml]: Lets remove the ignore. I'd rather see the lint warning than not.

File: app/src/dev/AndroidManifest.xml [hunk 6]
```diff
@@ -4,7 +4,8 @@
 
     <application
         android:allowBackup="false"
-        tools:replace="android:allowBackup">
+        tools:replace="android:allowBackup"
+        tools:ignore="GoogleAppIndexingWarning">
```
Comment (patrickhammond) [inline app/src/dev/AndroidManifest.xml]: Remove ignore

File: app/src/main/AndroidManifest.xml [hunk 7]
```diff
@@ -1,38 +1,41 @@
...
     <application
+        android:name=".app.MainApplication"
         android:allowBackup="true"
+        android:fullBackupContent="@xml/backup_rules"
         android:icon="@mipmap/ic_launcher"
         android:label="@string/app_name"
-        android:name=".app.MainApplication"
         android:theme="@style/AppTheme"
-        android:fullBackupContent="@xml/backup_rules">
+        tools:ignore="GoogleAppIndexingWarning">
```
Comment (patrickhammond) [inline app/src/main/AndroidManifest.xml]: Remove ignore

File: app/src/main/java/com/mycompany/myapp/ui/main/MainFragment.kt [hunk 8]
```diff
@@ -37,7 +37,7 @@ class MainFragment : BaseFragment() {
         binding = DataBindingUtil.inflate(inflater, R.layout.fragment_main, container, false)
         binding.vm = viewModel
 
-        binding.commits.layoutManager = LinearLayoutManager(activity)
+        binding.commits.layoutManager = androidx.recyclerview.widget.LinearLayoutManager(activity)
```
Comment (patrickhammond) [inline app/src/main/java/com/mycompany/myapp/ui/main/MainFragment.kt]: FQN

File: app/src/main/java/com/mycompany/myapp/ui/main/MainFragment.kt [hunk 9]
```diff
@@ -63,7 +63,7 @@ class MainFragment : BaseFragment() {
 
     private class CommitViewHolder(
             private val binding: CommitItemBinding)
-        : RecyclerView.ViewHolder(binding.root) {
+        : androidx.recyclerview.widget.RecyclerView.ViewHolder(binding.root) {
```
Comment (patrickhammond) [inline app/src/main/java/com/mycompany/myapp/ui/main/MainFragment.kt]: FQN

File: app/src/main/java/com/mycompany/myapp/util/databinding/RecyclerViewAdapters.kt [hunk 10]
```diff
@@ -13,9 +13,9 @@ object RecyclerViewAdapters {
      */
     @JvmStatic
     @BindingAdapter("items")
-    fun setItems(view: RecyclerView, items: List<Any>) {
+    fun setItems(view: androidx.recyclerview.widget.RecyclerView, items: List<Any>) {
```
Comment (patrickhammond) [inline app/src/main/java/com/mycompany/myapp/util/databinding/RecyclerViewAdapters.kt]: FQN

File: app/src/main/java/com/mycompany/myapp/util/recyclerview/ArrayAdapter.kt [hunk 11]
```diff
@@ -1,10 +1,10 @@
 package com.mycompany.myapp.util.recyclerview
 
-import android.support.v7.widget.RecyclerView
-import android.support.v7.widget.RecyclerView.ViewHolder
+import androidx.recyclerview.widget.RecyclerView
+import androidx.recyclerview.widget.RecyclerView.ViewHolder
 import java.util.*
 
-abstract class ArrayAdapter<Item, ItemViewHolder : ViewHolder> : RecyclerView.Adapter<ItemViewHolder>() {
+abstract class ArrayAdapter<Item, ItemViewHolder : ViewHolder> : androidx.recyclerview.widget.RecyclerView.Adapter<ItemViewHolder>() {
```
Comment (patrickhammond) [inline app/src/main/java/com/mycompany/myapp/util/recyclerview/ArrayAdapter.kt]: FQN

**Comment by akhilamadari** [comment]
Pushed with all the changes

**Comment by patrickhammond** [comment]
Fantastic!

**Pull request merged**